libass/ass_bitmap.c

```diff
@@ -42,7 +42,7 @@
 #undef ALIGN
 #undef DECORATE
 
-#if (defined(__i386__) || defined(__x86_64__)) && CONFIG_ASM
+#if CONFIG_ASM && CONFIG_X86
```

Comment:
I think it's better to mention that `CONFIG_X86` implies `CONFIG_ASM` and simplify condition. Also, do we really need `CONFIG_ASM`? Currently it's synonym to `CONFIG_X86` but even in the future we'd use something like `CONFIG_ARM` instead.

Reply:
Today, `CONFIG_X86` (or the corresponding Automake check) does not actually imply `CONFIG_ASM` if we found NASM but it turned out to be unusable or too old. But we could, of course, change that.

Using `CONFIG_<arch>` to mean “<arch> _asm_ is enabled” seems unintuitive to me. But we could, of course, rename this to `CONFIG_<arch>_ASM`.

Beyond that, I think it’s a coin toss. Maybe we’ll have some use for an arch-independent `CONFIG_ASM` section some day, maybe not.

Reply:
Fwiw having `CONFIG_ASM` distinct from the arch-macros seems more intuitive to me, but I don't care strongly about it.

Reply:
FYI: corresponding macro in [checkasm](https://github.com/libass/libass/pull/400/commits/c8c46a34115b18f6d7e341875441c4d797f90b37#diff-e3e4c2b9b20b19afe5d5359e81ef7f883576b28089ca959cdc811e8918842686L118) is called `ARCH_X86`. We can adopt that naming scheme to reduce future changeset.

Reply:
I don’t see `ARCH_X86` in the checkasm PR any more. I do see `ARCH_X86_64`, but it already exists, and it’s an assembler-only macro. So in a sense the `ARCH_` and `CONFIG_` macros are parallel: one for assembler and one for C. Maybe not optimal, but the prospect of using a non-`CONFIG_` name for just one out of all configure macros feels weird.

Reply:
It's in the original [checkasm](https://github.com/libass/libass/commit/3906012d9cf4afc19e522a7de749847657d0fbeb#diff-e3e4c2b9b20b19afe5d5359e81ef7f883576b28089ca959cdc811e8918842686L118), it got removed later in the PR.